packages/core/types/common.d.ts

```diff
@@ -46,11 +46,7 @@ export interface IFinalConfig extends IConfig {
   logger?: ILogger | null;
 }
 
-type SyncBeforeSend = (report: Report) => void;
-type AsyncBeforeSend = (report: Report, cb: (err: null | Error) => void) => void;
-type PromiseBeforeSend = (report: Report) => Promise<void>;
-
-export type BeforeSend = SyncBeforeSend | AsyncBeforeSend | PromiseBeforeSend;
+export type BeforeSend = (report: Report, cb?: (err: null | Error) => void) => void | Promise<void>;
```

Comment:
I'm not a big fan of this change, I liked the way it was before. However, it prevented the compiler from being able to implying the type of this function. Now this code would be allowed where it wouldn't before (allowing the mixing of callbacks and promises):

```typescript
beforeSend: (report, callback) => return new Promise(...)
```